python/ray/data/_internal/stats.py

```diff
@@ -310,94 +323,204 @@ def _summarize_iter(self) -> str:
             )
 
+        memory_stats = stage_stats.get("memory", {})
+        if memory_stats:
             out += indent
-            memory_stats = [
-                round(e.max_rss_bytes / (1024 * 1024), 2) for e in exec_stats
-            ]
             out += "* Peak heap memory usage (MiB): {} min, {} max, {} mean\n".format(
-                min(memory_stats),
-                max(memory_stats),
-                int(np.mean(memory_stats)),
+                memory_stats["min"],
+                memory_stats["max"],
+                memory_stats["mean"],
             )
 
-        output_num_rows = [m.num_rows for m in blocks if m.num_rows is not None]
-        if output_num_rows:
+        output_num_rows_stats = stage_stats.get("output_num_rows", {})
+        if output_num_rows_stats:
             out += indent
             out += "* Output num rows: {} min, {} max, {} mean, {} total\n".format(
-                min(output_num_rows),
-                max(output_num_rows),
-                int(np.mean(output_num_rows)),
-                sum(output_num_rows),
+                output_num_rows_stats["min"],
+                output_num_rows_stats["max"],
+                output_num_rows_stats["mean"],
+                output_num_rows_stats["sum"],
             )
 
-        output_size_bytes = [m.size_bytes for m in blocks if m.size_bytes is not None]
-        if output_size_bytes:
+        output_size_bytes_stats = stage_stats.get("output_size_bytes", {})
+        if output_size_bytes_stats:
             out += indent
             out += "* Output size bytes: {} min, {} max, {} mean, {} total\n".format(
-                min(output_size_bytes),
-                max(output_size_bytes),
-                int(np.mean(output_size_bytes)),
-                sum(output_size_bytes),
+                output_size_bytes_stats["min"],
+                output_size_bytes_stats["max"],
+                output_size_bytes_stats["mean"],
+                output_size_bytes_stats["sum"],
             )
 
-        if exec_stats:
-            node_counts = collections.defaultdict(int)
-            for s in exec_stats:
-                node_counts[s.node_id] += 1
+        node_count_stats = stage_stats.get("node_count", {})
+        if node_count_stats:
             out += indent
             out += "* Tasks per node: {} min, {} max, {} mean; {} nodes used\n".format(
-                min(node_counts.values()),
-                max(node_counts.values()),
-                int(np.mean(list(node_counts.values()))),
-                len(node_counts),
+                node_count_stats["min"],
+                node_count_stats["max"],
+                node_count_stats["mean"],
+                node_count_stats["count"],
             )
-
         return out
 
+    def get_total_wall_time(self) -> float:
+        total_wall_time = 0
+        for stage_uuid, ss in self.stage_stats.items():
+            stage_wall_time_stats = ss.get("wall_time")
```

Comment:
can we define all stats name (`wall_time`, ` memory`, etc) as variable and resue them here?